### ISSUE TITLE
TypeError due to having no spa

### ISSUE THREAD

**Issue by ProEdge2** [schemers/homebridge-screenlogic#4]
![220654FC-9A16-43FF-AC5F-43F4CE2F3529](https://user-images.githubusercontent.com/63991231/83215831-d6f5e800-a135-11ea-875e-ef91cba50b8b.png)

I keep on getting this error message and I’m unable to get the plugin to work, any advice?

**Comment by schemers** [comment]
It looks like the `isSpaActive()` function is failing. Do you have a Spa? If not I’ll have to add some logic to detect that and not access any spa-related functions.

**Comment by ProEdge2** [comment]
I do not have a spa would you be able to fix that?

**Comment by schemers** [comment]
I just published v1.4.2 which should bypass any spa-related logic if no spa is detected (as well as any pool-related logic if no pool is detected). please give it a shot and see if it fixes the issue.

**Comment by ProEdge2** [comment]
Ok thank you so much I will try it right now

**Comment by ProEdge2** [comment]
I just tried it now and it works, just a minor issue I tried switching it to fahrenheit but it stays in Celsius. Is there any way to fix that?

**Comment by schemers** [comment]
did you try switching using the "Hardware Display/Unit" setting on the accessory inside of HomeKit?

I've found that doesn't seem to work, even when using the HomeKit Accessory Simulator app from Apple.

The Home app seems to only honor the system-wide C/F setting, which is in Settings/General/Language & Region/Temperature Unit.

You can also toggle that setting in the Weather app by tapping on C/F in the lower left corner.

See also: https://www.reddit.com/r/ecobee/comments/axtcl3/wrong_temperature_unit_in_homekit/

**Comment by ProEdge2** [comment]
I’ve tried changing in within the hardware setting and no luck. Its ok Celsius is fine. Thanks for your help!

**Comment by schemers** [comment]
Out of curiosity, I did a little more research. It turns out that the Hardware Display/Unit setting is actually meant to change the units on the hardware itself, not in the home app.

I verified this by changing the hardware display units on my Nest thermostat in the Home app, and then checked on the Nest itself, and the units it was displaying switched to the new value. 

The units that get displayed on the iOS device are always controlled by the system-wide preference.

Theoretically I should be propagating the hardware display units back to the Pentair controller (which would then change the units displayed on any Pentair devices/apps), but I don't think the node module I'm using supports that, and it could potentially be confusing (like it was for both of us).
